src/frontend/apps/desk/src/features/members/components/ModalAddMembers.tsx

```diff
@@ -30,7 +38,53 @@ export const ModalAddMembers = ({
 }: ModalAddMembersProps) => {
   const { t } = useTranslation();
   const [selectedMembers, setSelectedMembers] = useState<OptionSelect>([]);
-  const [, setSelectedRole] = useState<Role>(Role.MEMBER);
+  const [selectedRole, setSelectedRole] = useState<Role>(Role.MEMBER);
+  const { toast } = useToastProvider();
+  const { mutateAsync: createInvitation } = useCreateInvitation();
+
+  const handleValidate = async () => {
+    const promises = await Promise.allSettled<Invitation>(
+      selectedMembers.map((selectedMember) => {
+        return new Promise<Invitation>((resolve, reject) => {
+          createInvitation({
+            email: selectedMember.value.email,
+            role: selectedRole,
+            teamId: team.id,
+          })
+            .then((data) => {
+              resolve(data);
+            })
+            .catch((error) => {
+              reject(error);
+            });
+        });
+      }),
+    );
```

Comment:
Could we avoid mixing `await` and `then`, I feel, and that personal, it makes the code hard to read.
I haven't tested my suggestion soooooo, there is a big chance that it's buggy, but wdyt of ? 

```
      const promises = selectedMembers.map(async (selectedMember) => {
        try {
          const data = await createInvitation({
            email: selectedMember.value.email,
            role: selectedRole,
            teamId: team.id,
          });
          return data;
        } catch (error) {
          throw error;
        }
      });

      const results = await Promise.allSettled(promises);
```

Reply:
Yes could be improved 👍 
https://github.com/numerique-gouv/people/pull/131/commits/c68b2b521cb64de4cb2d98d175dd7beb7249ea75